repose-aggregator/components/services/http-logging-service/http-logging-service-impl/src/main/scala/org/openrepose/core/services/httplogging/HttpLoggingConfigListener.scala

```diff
@@ -0,0 +1,111 @@
+/*
+ * _=_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_=
+ * Repose
+ * _-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-
+ * Copyright (C) 2010 - 2015 Rackspace US, Inc.
+ * _-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * =_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_=_
+ */
+package org.openrepose.core.services.httplogging
+
+import com.fasterxml.jackson.core.JsonParseException
+import com.typesafe.scalalogging.slf4j.StrictLogging
+import javax.inject.Named
+import org.jtwig.environment.{EnvironmentConfiguration, EnvironmentConfigurationBuilder}
+import org.jtwig.value.Undefined
+import org.jtwig.value.convert.string.{DefaultStringConverter, StringConverter}
+import org.jtwig.{JtwigModel, JtwigTemplate}
+import org.openrepose.commons.config.manager.UpdateListener
+import org.openrepose.core.services.httplogging.config.{Format, HttpLoggingConfig, Message}
+import org.openrepose.core.services.httplogging.jtwig.HttpLoggingEnvironmentConfiguration
+import org.slf4j.LoggerFactory
+import play.api.libs.json.Json
+
+import scala.collection.JavaConverters._
+import scala.util.{Success, Try}
+
+/**
+  * Handles configuration for the [[HttpLoggingService]].
+  */
+@Named
+class HttpLoggingConfigListener extends UpdateListener[HttpLoggingConfig] with StrictLogging {
+
+  import HttpLoggingConfigListener._
+
+  private var initialized: Boolean = false
+  private var templates: List[LoggableTemplate] = List.empty
+
+  override def configurationUpdated(configurationObject: HttpLoggingConfig): Unit = {
+    logger.trace("Updating HTTP Logging Service configuration")
+
+    templates = configurationObject.getMessage.asScala
+      .toList
+      .flatMap { message =>
+        Option(HttpLoggingEnvironmentConfiguration(message.getFormat))
+          .filter(
+            validateFormat(message, _)
+              .recover { case e: JsonParseException =>
+                logger.warn("Unable to validate JSON for {}", message, e)
+              }
+              .isSuccess)
+          .map(JtwigTemplate.inlineTemplate(message.getValue, _))
```

Comment:
We should probably make a note that leading and trailing whitespace is preserved from the config so extra lines will be in the log.
Or we could do a .trim here.

Reply:
I wrote up my thoughts on this in a less-public comment on the story:
https://repose.atlassian.net/browse/REP-7786?focusedCommentId=38519&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-38519

To be clear, leading and trailing whitespace is not preserved in all cases. In general, that is determined by the XML processor, the `xml:space` attribute, and the use of `<![CDATA[...]]>`. For our usage, we theoretically know what XML process is being used and how it is configured, so we should only have to worry about `xml:space` and `<![CDATA[...]]>`. Correct me if I'm wrong, but I believe it is that case that if `CDATA` is not used, leading and trailing whitespace will be removed.

Of course, there are many mechanisms already in place for dealing with whitespace. There are those described in the previous paragraph, but there is also [JTwig whitespace control](http://jtwig.org/documentation/reference/syntax/code-islands#whitespace-control) and the [JTwig trim function](http://jtwig.org/documentation/reference/functions#trim). And of course, the operator can always write message content using exactly the whitespace that they desire.

With all of those tools existing to handle whitespace and with outstanding questions about how to handle whitespace compression or elimination, I did not feel the need to either `trim` here or include a `crush` feature on the message content as part of this story.